lib/acme/certificate.rb

```diff
@@ -9,6 +9,39 @@ def initialize(certificate, chain, request)
     @x509 = certificate
     @x509_chain = chain
     @request = request
+
+    #create_self_signed_cert if x509.nil?
+  end
+
+  def create_self_signed_cert subjectAltDomains
+    keypair = OpenSSL::PKey::RSA.new(2048) 
+
+    cert = OpenSSL::X509::Certificate.new
+    cert.subject = cert.issuer = OpenSSL::X509::Name.parse "/CN=#{subjectAltDomains.first}"
+    cert.not_before = Time.now
+    cert.not_after = Time.now + 48*3600
+    cert.public_key = keypair.public_key
+    cert.version = 2 # X509v3
+    cert.serial = 0x0
+
+    ext = []
+
+    ef = OpenSSL::X509::ExtensionFactory.new
+    ef.subject_certificate = cert
+    ef.issuer_certificate = cert
+
+    cert.extensions = [
+        ef.create_extension("basicConstraints","CA:TRUE", true),
```

Comment:
The official client also sets `pathlen:0` here. https://github.com/letsencrypt/letsencrypt/blob/master/acme/acme/crypto_util.py#L212